LPD-26897 Data Set Selector in fragment doesn't keep the selected data set

markocikos opened the pull request:
### References

- JIRA: [LPD-26897 Data Set Selector in fragment doesn't keep the selected data set](https://issues.liferay.com/browse/LPD-26897)
- This is a followup on https://github.com/liferay-frontend/liferay-portal/pull/4362

### What is the goal of this PR

We're adding missing capability to selection modal that enables working with selection data to implementations other than search container. We're then applying this to item selector of Data Set fragment.

Few technical notes in comments.

### What does it look like?

https://github.com/user-attachments/assets/ef96cea0-2a14-43e0-a51a-3be21f081f4f

![image](https://github.com/user-attachments/assets/c9cc3407-aba4-4299-9b51-2f0668e8064e)

### Steps to reproduce

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-37.liferay.com/job/test-portal-source-format/9998/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/9998/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3">e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPD-26897">LPD-26897</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/f3524893ac55e08b7457763e82c3857c6dfb38f3">f3524893ac55e08b7457763e82c3857c6dfb38f3</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-37.liferay.com/job/test-portal-source-format/9998/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/9998/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-37.liferay.com/job/test-portal-source-format/9998/">test-portal-source-format#9998</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/9998/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/4382">liferay-frontend#4382</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/34754/routines/34756">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/34754/routines/34756/build/65424745">[master] ci:test:sf - markocikos &gt; liferay-frontend - PR#4382 - 2024-08-19[06:49:55]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>65424745<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-37.liferay.com/job/publish-testray-report/38264/">publish-testray-report#38264</a><br/></div></div>

markocikos commented on modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/item/selector/FDSAdminItemSelector.tsx:
```diff
@@ -35,11 +35,31 @@ const FDSAdminItemSelector = ({
 }) => {
-	const [selectedItem, setSelectedItem] = useState<ISelectedItem>();
+	const getSelectedData = () => {
+		const dataset = (window.frameElement as HTMLElement)?.dataset;
+
+		const externalReferenceCode = dataset.selecteditemsercs;
+		const id = dataset.selecteditemsids;
+		const label = dataset.selecteditemslabels;
```
The central problem of this task was how to wire selected data to FDS. There are several ways to do it, but I liked the solution from @juanjofgliferay 's previous iteration to use iframe HTML. I only changed to `data-*` attributes, as those better fit the purpose.

Notes:
- The [existing solution](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L541-L622) in `openSelectionModal`, wiring `onOpen`, was always intended to work specifically with search container. It does not fit React. We could force an ugly solution, but there is no need. Iframe attribute is better as we know it'll be set before React resolves.
- Alternative to iframe attributes, that I strongly considered, is adding selection data to URL as parameters. This runs into problems with portlet namespacing, would require more modifications of `layout-content-page-editor`, and could have edge case weird complications. It would also result in more boilerplate, as params need to be handled in backend DisplayContext. It is a viable option is this turn out not acceptable for some reason.

markocikos commented on modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts:
```diff
@@ -348,14 +389,14 @@ test(
-				structuredContentDescription,
-				'',
-			]);
+			).toEqual(
+				expect.arrayContaining([
+					structuredContentTitle,
+					structuredContentDescription,
+				])
+			);
```
With this change, test works even if you (locally) have a web content.

markocikos commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/10975/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/10975/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:stable - 26 out of 27 jobs passed</h3><h3>:x: ci:test:relevant - 45 out of 49 jobs passed in 1 hour 35 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3">e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3">e69f21c3db55287fe5511bcadf6fd34cdfb5e5d3</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/#/project/34754/routines/174485/build/64463083">EE Development Acceptance (master) - 752 - 2024-08-16[20:38:16]</a></p><details><summary><strong>ci:test:stable - 26 out of 27 jobs PASSED</strong></summary><div><h4>1 Failed Jobs:</h4><ul><li>central-requirements-jdk8<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/432523/">central-requirements-jdk8/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1175525/">functional-smoke-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/367658/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420106/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/1</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1078236/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/2</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/934645/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/3</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/418916/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/4</a></li></ul></li><li>js-unit-jdk8<ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/458764/">js-unit-jdk8/0/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353240/"><strike>js-unit-jdk8/1/0</strike></a></li></ul></li><li>modules-compile-jdk8<ul><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420104/">modules-compile-jdk8/0/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420105/">modules-compile-jdk8/0/1</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/418915/">modules-compile-jdk8/0/2</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/412360/">modules-compile-jdk8/0/3</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1175527/">modules-compile-jdk8/0/4</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1014324/">modules-compile-jdk8/0/5</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/412361/">modules-compile-jdk8/0/6</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/658759/">modules-compile-jdk8/0/7</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/460935/">modules-compile-jdk8/0/8</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/460936/">modules-compile-jdk8/0/9</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/418914/">modules-compile-jdk8/0/10</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353238/">modules-compile-jdk8/0/11</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/658758/">modules-compile-jdk8/0/12</a></li></ul></li><li>modules-integration-mysql57-jdk8_stable<ul><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/934644/">modules-integration-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930055/">modules-integration-mysql57-jdk8_stable/0/1</a></li></ul></li><li>modules-unit-jdk8_stable<ul><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353239/">modules-unit-jdk8_stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1068830/">playwright-js-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>unit-jdk8_stable<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959766/">unit-jdk8_stable/0/0</a></li></ul></li></ul></div><details><summary><strong>26 Successful Jobs:</strong></summary><ul/></details></details><details><summary><strong>ci:test:relevant - 45 out of 49 jobs PASSED</strong></summary><div><h4>4 Failed Jobs:</h4><ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/10975/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>central-requirements-jdk8<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/432523/">central-requirements-jdk8/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1175525/">functional-smoke-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/367658/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420106/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/1</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1078236/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/2</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/934645/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/3</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/418916/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/4</a></li></ul></li><li>js-unit-jdk8<ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/458764/">js-unit-jdk8/0/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353240/"><strike>js-unit-jdk8/1/0</strike></a></li></ul></li><li>modules-compile-jdk8<ul><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420104/">modules-compile-jdk8/0/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420105/">modules-compile-jdk8/0/1</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/418915/">modules-compile-jdk8/0/2</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/412360/">modules-compile-jdk8/0/3</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1175527/">modules-compile-jdk8/0/4</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1014324/">modules-compile-jdk8/0/5</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/412361/">modules-compile-jdk8/0/6</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/658759/">modules-compile-jdk8/0/7</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/460935/">modules-compile-jdk8/0/8</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/460936/">modules-compile-jdk8/0/9</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/418914/">modules-compile-jdk8/0/10</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353238/">modules-compile-jdk8/0/11</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/658758/">modules-compile-jdk8/0/12</a></li></ul></li><li>modules-integration-mysql57-jdk8<ul><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/896681/">modules-integration-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1078234/">modules-integration-mysql57-jdk8/0/1</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1068831/">modules-integration-mysql57-jdk8/0/2</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/953450/">modules-integration-mysql57-jdk8/0/3</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/934643/">modules-integration-mysql57-jdk8/0/4</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962047/">modules-integration-mysql57-jdk8/0/5</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1175526/">modules-integration-mysql57-jdk8/0/6</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1068829/">modules-integration-mysql57-jdk8/0/7</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1078235/">modules-integration-mysql57-jdk8/0/8</a></li><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959769/">modules-integration-mysql57-jdk8/0/9</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962048/">modules-integration-mysql57-jdk8/0/10</a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1027351/">modules-integration-mysql57-jdk8/0/11</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962049/">modules-integration-mysql57-jdk8/0/12</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/985081/">modules-integration-mysql57-jdk8/0/13</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930056/">modules-integration-mysql57-jdk8/0/14</a></li></ul></li><li>modules-integration-mysql57-jdk8_stable<ul><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/934644/">modules-integration-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930055/">modules-integration-mysql57-jdk8_stable/0/1</a></li></ul></li><li>modules-semantic-versioning-jdk8<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959767/">modules-semantic-versioning-jdk8/0/0</a></li></ul></li><li>modules-unit-jdk8<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959768/">modules-unit-jdk8/0/0</a></li></ul></li><li>modules-unit-jdk8_stable<ul><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353239/">modules-unit-jdk8_stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/985080/">playwright-js-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/359964/">playwright-js-tomcat90-mysql57-jdk8/1/0</a></li><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770/"><strike>playwright-js-tomcat90-mysql57-jdk8/2/0</strike></a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/432524/"><strike>playwright-js-tomcat90-mysql57-jdk8/3/0</strike></a></li></ul></li><li>playwright-js-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1068830/">playwright-js-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>unit-jdk8_stable<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959766/">unit-jdk8_stable/0/0</a></li></ul></li></ul></div><details><summary><strong>45 Successful Jobs:</strong></summary><ul/></details></details><h5>For more details click <a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/10975/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353240//consoleText">js-unit-jdk8/1/0</a><ol><li><div><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353240//testReport/layout.layout-content-page-editor-web.test.page_editor/common/openItemSelector_calls_destroyCallback_on_modal_close_event">layout.layout-content-page-editor-web.test.page_editor.common.openItemSelector calls destroyCallback on modal close event</a><pre>TypeError: Cannot read properties of undefined (reading 'externalReferenceCode')
    at openItemSelector (layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/openItemSelector.js:56:41)
    at openModal (layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js:17:2)
    at Object.&lt;anonymous&gt; (layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js:39:3)
    at Object.asyncJestTest (/opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
    at /opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:43:12
    at new Promise (&lt;anonymous&gt;)
    at mapper (/opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at /opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:73:41
    at processTicksAndRejections (node:internal/process/task_queues:95:5)</pre></div></li><li><div><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353240//testReport/layout.layout-content-page-editor-web.test.page_editor/common/openItemSelector_extracts_first_item_if_selection_does_not_have_value">layout.layout-content-page-editor-web.test.page_editor.common.openItemSelector extracts first item if selection does not have value</a><pre>TypeError: Cannot read properties of undefined (reading 'externalReferenceCode')
    at openItemSelector (layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/openItemSelector.js:56:41)
    at openModal (layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js:17:2)
    at Object.&lt;anonymous&gt; (layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js:105:22)
    at Object.asyncJestTest (/opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
    at /opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:43:12
    at new Promise (&lt;anonymous&gt;)
    at mapper (/opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at /opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:73:41
    at processTicksAndRejections (node:internal/process/task_queues:95:5)</pre></div></li><li><div><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353240//testReport/layout.layout-content-page-editor-web.test.page_editor/common/openItemSelector_injects_selection_value_into_infoItem">layout.layout-content-page-editor-web.test.page_editor.common.openItemSelector injects selection.value into infoItem</a><pre>TypeError: Cannot read properties of undefined (reading 'externalReferenceCode')
    at openItemSelector (layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/openItemSelector.js:56:41)
    at openModal (layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js:17:2)
    at Object.&lt;anonymous&gt; (layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js:60:22)
    at Object.asyncJestTest (/opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
    at /opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:43:12
    at new Promise (&lt;anonymous&gt;)
    at mapper (/opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at /opt/dev/projects/github/liferay-portal/modules/node_modules/jest-jasmine2/build/queueRunner.js:73:41
    at processTicksAndRejections (node:internal/process/task_queues:95:5)</pre></div></li><li>...</li></ol></div></li><li><div><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770//consoleText">playwright-js-tomcat90-mysql57-jdk8/2/0</a> - <a href="https://testray.liferay.com/reports/production/logs/2024-08/test-1-39/test-portal-acceptance-pullrequest(master)/10975/playwright-js-tomcat90-mysql57-jdk8/2/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770/testReport/layout-content-page-editor-web_fragments.spec/ts">layout-content-page-editor-web/fragments.spec.ts</a></summary><div><ol><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770//testReport/layout-content-page-editor-web_fragments.spec/ts/Multiselect_Fragment___Allow_submit_form_if_the_field_is_required_and_at_least_one_item_is_checked">Multiselect Fragment › Allow submit form if the field is required and at least one item is checked</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770/testReport/layout-content-page-editor-web_collectionFilterFragment.spec/ts">layout-content-page-editor-web/collectionFilterFragment.spec.ts</a></summary><div><ol><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770//testReport/layout-content-page-editor-web_collectionFilterFragment.spec/ts/Enables_search_field_in_dropdown_list_of_Collection_Filter">Enables search field in dropdown list of Collection Filter</a></li><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770//testReport/layout-content-page-editor-web_collectionFilterFragment.spec/ts/Filters_a_web_content_collection_by_single_and_multiple_categories">Filters a web content collection by single and multiple categories</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770/testReport/layout-content-page-editor-web_mapping.spec/ts">layout-content-page-editor-web/mapping.spec.ts</a></summary><div><ol><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/959770//testReport/layout-content-page-editor-web_mapping.spec/ts/Allows_selecting_specific_repeatable_collection_provider">Allows selecting specific repeatable collection provider</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/432524//consoleText">playwright-js-tomcat90-mysql57-jdk8/3/0</a> - <a href="https://testray.liferay.com/reports/production/logs/2024-08/test-1-39/test-portal-acceptance-pullrequest(master)/10975/playwright-js-tomcat90-mysql57-jdk8/3/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/432524/testReport/frontend-data-set-admin-web_tests_data-set-admin_dataSets.spec/ts">frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts</a></summary><div><ol><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/432524//testReport/frontend-data-set-admin-web_tests_data-set-admin_dataSets.spec/ts/Can_sort_Data_Set_by_different_columns">Can sort Data Set by different columns</a></li></ol></div></details></li></ol></div></li></ol><h4>For upstream results, click <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-39/jobs/test-portal-acceptance-pullrequest(master)/builds/10975/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-39/jobs/test-portal-acceptance-pullrequest(master)/builds/10975/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-39/jobs/test-portal-acceptance-pullrequest(master)/builds/10975/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/10975/">test-portal-acceptance-pullrequest(master)#10975</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/10975/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/4382">liferay-frontend#4382</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/34754/routines/34756">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/34754/routines/34756/build/65457188">[master] ci:test:relevant - markocikos &gt; liferay-frontend - PR#4382 - 2024-08-19[07:14:28]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>65457188<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-39.liferay.com/job/publish-testray-report/23570/">publish-testray-report#23570</a><br/></div></div>

juanjofgliferay commented:
@markocikos LGTM :rocket: 

We could add a simple test to check that we can pass data to the modal in the https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/openItemSelector.test.js#L17

In this test file test are only checking individual props, so something like this will be enough:

```


         it('passes selectedItem information to the modal', () => {
		const callback = jest.fn();
		const {onSelect} = openModal({callback});

		const item = {
			classPK: '',
			externalReferenceCode: 'someERC',
			title: '',
		};

		onSelect({selectedItem: item});

		expect(callback).toHaveBeenCalledWith(item);
	});
```

juanjofgliferay reviewed (approved):
:tada:

markocikos commented:
Pushed! Changes:
- added unit tests for `openItemSelector`
- fixed undefined handling that broke tests in https://github.com/liferay-frontend/liferay-portal/pull/4382#issuecomment-2296916076

There are still two failures in `layout-content-page-editor-web` locally, but they don't seem related to me:
![image](https://github.com/user-attachments/assets/8e9adc62-cd3c-48fb-bd0b-0c405eb5009f)

markocikos commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-26.liferay.com/job/test-portal-source-format/8149/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-source-format/builds/8149/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:sf - 0 out of 1 jobs passed in 2 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/94279c2413017891a3bfada7d82bb4d61575bf68">94279c2413017891a3bfada7d82bb4d61575bf68</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPD-26897">LPD-26897</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/e34eee65f039383f5f0463cb460975b4d4cc0903">e34eee65f039383f5f0463cb460975b4d4cc0903</a></p>0 out of 1jobs PASSED<div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-26.liferay.com/job/test-portal-source-format/8149/"><strike>test-portal-source-format</strike></a></li></ul></div><h5>For more details click <a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-source-format/builds/8149/jenkins-report.html">here</a>.</h5><pre>  at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
  at org.apache.tools.ant.Task.perform(Task.java:350)
  at net.sf.antcontrib.logic.TryCatchTask.execute(TryCatchTask.java:207)
  at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
  at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
  at org.apache.tools.ant.Task.perform(Task.java:350)
  at org.apache.tools.ant.Target.execute(Target.java:449)
  at org.apache.tools.ant.Target.performTasks(Target.java:470)
  at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
  at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
  at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
  at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
  at org.apache.tools.ant.Main.runBuild(Main.java:818)
  at org.apache.tools.ant.Main.startAnt(Main.java:223)
  at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
  at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
[beanshell] Created cache file in /tmp/jenkins-cached-files/418342683.txt

BUILD FAILED
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-common.xml:15471: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-test-portal-source-format.xml:121: Sourced file: inline evaluation of: ``                import com.liferay.jenkins.results.parser.AntUtil;         import com.lifer . . . '' : TargetError : at Line: 211 : in file: inline evaluation of: ``                import com.liferay.jenkins.results.parser.AntUtil;         import com.lifer . . . '' : throw e ; 

Target exception: java.lang.RuntimeException: The following issues were found:
1. Unable to create batch modules-unit-jdk8 since modules.includes.required.test.batch.class.names.includes[layout-rule][relevant][modules-unit-jdk8] is not set in /opt/dev/projects/github/liferay-portal/modules/apps/layout/test.properties
2. Unable to create batch modules-unit-jdk8 since modules.includes.required.test.batch.class.names.includes[layout-content-page-editor-web-rule][relevant][modules-unit-jdk8] is not set in /opt/dev/projects/github/liferay-portal/modules/apps/layout/layout-content-page-editor-web/test.properties</pre></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-26.liferay.com/job/test-portal-source-format/8149/">test-portal-source-format#8149</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-source-format/builds/8149/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/4382">liferay-frontend#4382</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/34754/routines/34756">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/34754/routines/34756/build/66460244">[master] ci:test:sf - markocikos &gt; liferay-frontend - PR#4382 - 2024-08-20[05:37:34]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>66460244<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-26.liferay.com/job/publish-testray-report/32176/">publish-testray-report#32176</a><br/></div></div>

markocikos commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-35.liferay.com/job/test-portal-source-format/3044/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-source-format/builds/3044/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/4da60e29fa13a98180e553f79b398d7c46634b41">4da60e29fa13a98180e553f79b398d7c46634b41</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPD-26897">LPD-26897</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/e34eee65f039383f5f0463cb460975b4d4cc0903">e34eee65f039383f5f0463cb460975b4d4cc0903</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-35.liferay.com/job/test-portal-source-format/3044/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-source-format/builds/3044/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/5873/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5873/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 27 out of 27 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 49 out of 49 jobs passed in 1 hour 20 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/4da60e29fa13a98180e553f79b398d7c46634b41">4da60e29fa13a98180e553f79b398d7c46634b41</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2723417e0e6fc75711d921b2f6699531e7ba3ade">2723417e0e6fc75711d921b2f6699531e7ba3ade</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/#/project/34754/routines/174485/build/66378057">EE Development Acceptance (master) - 756 - 2024-08-20[00:43:16]</a></p><details><summary><strong>ci:test:stable - 27 out of 27 jobs PASSED</strong></summary><details><summary><strong>27 Successful Jobs:</strong></summary><ul><li>central-requirements-jdk8<ul><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930967/">central-requirements-jdk8/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/954628/">functional-smoke-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1079314/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897791/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/1</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353714/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/2</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/419363/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/3</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420630/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/4</a></li></ul></li><li>js-unit-jdk8<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960913/">js-unit-jdk8/0/0</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1079315/">js-unit-jdk8/1/0</a></li></ul></li><li>modules-compile-jdk8<ul><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420628/">modules-compile-jdk8/0/0</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/935413/">modules-compile-jdk8/0/1</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/433068/">modules-compile-jdk8/0/2</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1015282/">modules-compile-jdk8/0/3</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/954630/">modules-compile-jdk8/0/4</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1015283/">modules-compile-jdk8/0/5</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897790/">modules-compile-jdk8/0/6</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/412887/">modules-compile-jdk8/0/7</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/419362/">modules-compile-jdk8/0/8</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1058414/">modules-compile-jdk8/0/9</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897788/">modules-compile-jdk8/0/10</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420629/">modules-compile-jdk8/0/11</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/659832/">modules-compile-jdk8/0/12</a></li></ul></li><li>modules-integration-mysql57-jdk8_stable<ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/659833/">modules-integration-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/963042/">modules-integration-mysql57-jdk8_stable/0/1</a></li></ul></li><li>modules-unit-jdk8_stable<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1176461/">modules-unit-jdk8_stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/459276/">playwright-js-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>unit-jdk8_stable<ul><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420627/">unit-jdk8_stable/0/0</a></li></ul></li></ul></details></details><details><summary><strong>ci:test:relevant - 49 out of 49 jobs PASSED</strong></summary><details><summary><strong>49 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/5873/">test-portal-acceptance-pullrequest(master)</a></li><li>central-requirements-jdk8<ul><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930967/">central-requirements-jdk8/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/954628/">functional-smoke-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1079314/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897791/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/1</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353714/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/2</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/419363/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/3</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420630/">functional-upgrade-tomcat90-mysql57-jdk8_stable/0/4</a></li></ul></li><li>js-unit-jdk8<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960913/">js-unit-jdk8/0/0</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1079315/">js-unit-jdk8/1/0</a></li></ul></li><li>modules-compile-jdk8<ul><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420628/">modules-compile-jdk8/0/0</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/935413/">modules-compile-jdk8/0/1</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/433068/">modules-compile-jdk8/0/2</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1015282/">modules-compile-jdk8/0/3</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/954630/">modules-compile-jdk8/0/4</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1015283/">modules-compile-jdk8/0/5</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897790/">modules-compile-jdk8/0/6</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/412887/">modules-compile-jdk8/0/7</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/419362/">modules-compile-jdk8/0/8</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1058414/">modules-compile-jdk8/0/9</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897788/">modules-compile-jdk8/0/10</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420629/">modules-compile-jdk8/0/11</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/659832/">modules-compile-jdk8/0/12</a></li></ul></li><li>modules-integration-mysql57-jdk8<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1176462/">modules-integration-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/935416/">modules-integration-mysql57-jdk8/0/1</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1058416/">modules-integration-mysql57-jdk8/0/2</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930968/">modules-integration-mysql57-jdk8/0/3</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897789/">modules-integration-mysql57-jdk8/0/4</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1079312/">modules-integration-mysql57-jdk8/0/5</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/954629/">modules-integration-mysql57-jdk8/0/6</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1058413/">modules-integration-mysql57-jdk8/0/7</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/930966/">modules-integration-mysql57-jdk8/0/8</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/963040/">modules-integration-mysql57-jdk8/0/9</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1058415/">modules-integration-mysql57-jdk8/0/10</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/954631/">modules-integration-mysql57-jdk8/0/11</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/935415/">modules-integration-mysql57-jdk8/0/12</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/986050/">modules-integration-mysql57-jdk8/0/13</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1015284/">modules-integration-mysql57-jdk8/0/14</a></li></ul></li><li>modules-integration-mysql57-jdk8_stable<ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/659833/">modules-integration-mysql57-jdk8_stable/0/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/963042/">modules-integration-mysql57-jdk8_stable/0/1</a></li></ul></li><li>modules-semantic-versioning-jdk8<ul><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/897787/">modules-semantic-versioning-jdk8/0/0</a></li></ul></li><li>modules-unit-jdk8<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/433067/">modules-unit-jdk8/0/0</a></li></ul></li><li>modules-unit-jdk8_stable<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1176461/">modules-unit-jdk8_stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1079313/">playwright-js-tomcat90-mysql57-jdk8/0/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/353713/">playwright-js-tomcat90-mysql57-jdk8/1/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/963041/">playwright-js-tomcat90-mysql57-jdk8/2/0</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/935414/">playwright-js-tomcat90-mysql57-jdk8/3/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-jdk8_stable<ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/459276/">playwright-js-tomcat90-mysql57-jdk8_stable/0/0</a></li></ul></li><li>unit-jdk8_stable<ul><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/420627/">unit-jdk8_stable/0/0</a></li></ul></li></ul></details></details><h5>For more details click <a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5873/jenkins-report.html">here</a>.</h5></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/5873/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/5873/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/5873/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

markocikos commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
Skipping previously passed test suites:
- `ci:test:relevant`
- `ci:test:sf`
- `ci:test:stable`

liferay-continuous-integration commented:
All required test suite(s) passed.
Forwarding pull request to `brianchandotcom`.
[Console](https://test-1-38.liferay.com/job/forward-pullrequest/1169//consoleFull)

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/153440
[Console](https://test-1-38.liferay.com/job/forward-pullrequest/1169//consoleFull)
